src/components/ZclEndpointCard.vue

```diff
@@ -289,6 +300,19 @@ export default {
     }
   },
   methods: {
+    // Calculate readable category string
+    getDeviceCategory(packageRef) {
+      let category = ''
+      this.$store.state.zap.selectedZapConfig.zclProperties.forEach((item) => {
+        if (item.id === packageRef) {
+          if (item.category) {
+            category = item.category
+            return
```

Comment:
Just realized the returning in a forEach does not really end the loop but it continues going through the loop regardless.

Can you switch `this.$store.state.zap.selectedZapConfig.zclProperties.forEach((item) => {
         if (item.id === packageRef) {
           if (item.category) {
             category = item.category
             return
           }
         }
       })`
       
to

`
let zclProperty = this.$store.state.zap.selectedZapConfig.zclProperties.find((item) => (item.id === packageRef && item.category))
if (zclProperty) {
category = zclProperty.category
} 
return category
`

Reply:
You can also switch to a normal for loop and return instead of using forEach

Reply:
you are right, thanks. i will fix it